sqlite3_opt_serialize.go

```diff
@@ -0,0 +1,68 @@
+// +build !sqlite_omit_deserialize
+
+package sqlite3
+
+/*
+#ifndef USE_LIBSQLITE3
+#include <sqlite3-binding.h>
+#else
+#include <sqlite3.h>
+#endif
+#include <stdlib.h>
+#include <stdint.h>
+*/
+import "C"
+
+import (
+	"fmt"
+	"unsafe"
+)
+
+// Serialize returns a byte slice that is a serialization of the database.
+// If the database fails to serialize, a nil slice will be returned.
```

Comment:
Can we make it return an error in this case to play more nicely with the Go ecosystem? Otherwise clients that don't read this carefully might think it always succeeds.